tsdb/index/index.go

```diff
@@ -720,6 +724,8 @@ func (w *Writer) writePostingsOffsetTable() error {
 	}
 
 	// Cleanup temporary file.
+	f.Close() // Needed for Windows.
```

Comment:
Do we care about errors on close? (I know we used to not care) :thinking: should we log at least? It's a pretty unlikely error and not related to the bug we fix here, so not really necessary for this PR.

FYI we have this helper on Thanos: https://github.com/thanos-io/thanos/blob/865d5ec710b5651ead08a7dc73864a6f5dbeeaa9/pkg/runutil/runutil.go#L119

Reply:
We don't generally care, but I've returned it on the happy path here just in case.

Reply:
Thanks!